wget/1.19.1_1/share/man/man1/wget.1
.\" Automatically generated by Pod::Man 2.27 (Pod::Simple 3.28)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{
.    if \nF \{
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\" ========================================================================
.\"
.IX Title "WGET 1"
.TH WGET 1 "2017-06-16" "GNU Wget 1.19.1" "GNU Wget"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
Wget \- The non\-interactive network downloader.
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
wget [\fIoption\fR]... [\fI\s-1URL\s0\fR]...
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\s-1GNU\s0 Wget is a free utility for non-interactive download of files from
the Web.  It supports \s-1HTTP, HTTPS,\s0 and \s-1FTP\s0 protocols, as
well as retrieval through \s-1HTTP\s0 proxies.
.PP
Wget is non-interactive, meaning that it can work in the background,
while the user is not logged on.  This allows you to start a retrieval
and disconnect from the system, letting Wget finish the work.  By
contrast, most of the Web browsers require constant user's presence,
which can be a great hindrance when transferring a lot of data.
.PP
Wget can follow links in \s-1HTML, XHTML,\s0 and \s-1CSS\s0 pages, to
create local versions of remote web sites, fully recreating the
directory structure of the original site.  This is sometimes referred to
as \*(L"recursive downloading.\*(R"  While doing that, Wget respects the Robot
Exclusion Standard (\fI/robots.txt\fR).  Wget can be instructed to
convert the links in downloaded files to point at the local files, for
offline viewing.
.PP
Wget has been designed for robustness over slow or unstable network
connections; if a download fails due to a network problem, it will
keep retrying until the whole file has been retrieved.  If the server
supports regetting, it will instruct the server to continue the
download from where it left off.
.SH "OPTIONS"
.IX Header "OPTIONS"
.SS "Option Syntax"
.IX Subsection "Option Syntax"
Since Wget uses \s-1GNU\s0 getopt to process command-line arguments, every
option has a long form along with the short one.  Long options are
more convenient to remember, but take time to type.  You may freely
mix different option styles, or specify options after the command-line
arguments.  Thus you may write:
.PP
.Vb 1
\&        wget \-r \-\-tries=10 http://fly.srk.fer.hr/ \-o log
.Ve
.PP
The space between the option accepting an argument and the argument may
be omitted.  Instead of \fB\-o log\fR you can write \fB\-olog\fR.
.PP
You may put several options that do not require arguments together,
like:
.PP
.Vb 1
\&        wget \-drc <URL>
.Ve
.PP
This is completely equivalent to:
.PP
.Vb 1
\&        wget \-d \-r \-c <URL>
.Ve
.PP
Since the options can be specified after the arguments, you may
terminate them with \fB\-\-\fR.  So the following will try to download
\&\s-1URL \s0\fB\-x\fR, reporting failure to \fIlog\fR:
.PP
.Vb 1
\&        wget \-o log \-\- \-x
.Ve
.PP
The options that accept comma-separated lists all respect the convention
that specifying an empty list clears its value.  This can be useful to
clear the \fI.wgetrc\fR settings.  For instance, if your \fI.wgetrc\fR
sets \f(CW\*(C`exclude_directories\*(C'\fR to \fI/cgi\-bin\fR, the following
example will first reset it, and then set it to exclude \fI/~nobody\fR
and \fI/~somebody\fR.  You can also clear the lists in \fI.wgetrc\fR.
.PP
.Vb 1
\&        wget \-X " \-X /~nobody,/~somebody
.Ve
.PP
Most options that do not accept arguments are \fIboolean\fR options,
so named because their state can be captured with a yes-or-no
(\*(L"boolean\*(R") variable.  For example, \fB\-\-follow\-ftp\fR tells Wget
to follow \s-1FTP\s0 links from \s-1HTML\s0 files and, on the other hand,
\&\fB\-\-no\-glob\fR tells it not to perform file globbing on \s-1FTP\s0 URLs.  A
boolean option is either \fIaffirmative\fR or \fInegative\fR
(beginning with \fB\-\-no\fR).  All such options share several
properties.
.PP
Unless stated otherwise, it is assumed that the default behavior is
the opposite of what the option accomplishes.  For example, the
documented existence of \fB\-\-follow\-ftp\fR assumes that the default
is to \fInot\fR follow \s-1FTP\s0 links from \s-1HTML\s0 pages.
.PP
Affirmative options can be negated by prepending the \fB\-\-no\-\fR to
the option name; negative options can be negated by omitting the
\&\fB\-\-no\-\fR prefix.  This might seem superfluous\-\-\-if the default for
an affirmative option is to not do something, then why provide a way
to explicitly turn it off?  But the startup file may in fact change
the default.  For instance, using \f(CW\*(C`follow_ftp = on\*(C'\fR in
\&\fI.wgetrc\fR makes Wget \fIfollow\fR \s-1FTP\s0 links by default, and
using \fB\-\-no\-follow\-ftp\fR is the only way to restore the factory
default from the command line.
.SS "Basic Startup Options"
.IX Subsection "Basic Startup Options"
.IP "\fB\-V\fR" 4
.IX Item "-V"
.PD 0
.IP "\fB\-\-version\fR" 4
.IX Item "--version"
.PD
Display the version of Wget.
.IP "\fB\-h\fR" 4
.IX Item "-h"
.PD 0
.IP "\fB\-\-help\fR" 4
.IX Item "--help"
.PD
Print a help message describing all of Wget's command-line options.
.IP "\fB\-b\fR" 4
.IX Item "-b"
.PD 0
.IP "\fB\-\-background\fR" 4
.IX Item "--background"
.PD
Go to background immediately after startup.  If no output file is
specified via the \fB\-o\fR, output is redirected to \fIwget-log\fR.
.IP "\fB\-e\fR \fIcommand\fR" 4
.IX Item "-e command"
.PD 0
.IP "\fB\-\-execute\fR \fIcommand\fR" 4
.IX Item "--execute command"
.PD
Execute \fIcommand\fR as if it were a part of \fI.wgetrc\fR.  A command thus invoked will be executed
\&\fIafter\fR the commands in \fI.wgetrc\fR, thus taking precedence over
them.  If you need to specify more than one wgetrc command, use multiple
instances of \fB\-e\fR.
.SS "Logging and Input File Options"
.IX Subsection "Logging and Input File Options"
.IP "\fB\-o\fR \fIlogfile\fR" 4
.IX Item "-o logfile"
.PD 0
.IP "\fB\-\-output\-file=\fR\fIlogfile\fR" 4
.IX Item "--output-file=logfile"
.PD
Log all messages to \fIlogfile\fR.  The messages are normally reported
to standard error.
.IP "\fB\-a\fR \fIlogfile\fR" 4
.IX Item "-a logfile"
.PD 0
.IP "\fB\-\-append\-output=\fR\fIlogfile\fR" 4
.IX Item "--append-output=logfile"
.PD
Append to \fIlogfile\fR.  This is the same as \fB\-o\fR, only it appends
to \fIlogfile\fR instead of overwriting the old log file.  If
\&\fIlogfile\fR does not exist, a new file is created.
.IP "\fB\-d\fR" 4
.IX Item "-d"
.PD 0
.IP "\fB\-\-debug\fR" 4
.IX Item "--debug"
.PD
Turn on debug output, meaning various information important to the
developers of Wget if it does not work properly.  Your system
administrator may have chosen to compile Wget without debug support, in
which case \fB\-d\fR will not work.  Please note that compiling with
debug support is always safe\-\-\-Wget compiled with the debug support will
\&\fInot\fR print any debug info unless requested with \fB\-d\fR.
.IP "\fB\-q\fR" 4
.IX Item "-q"
.PD 0
.IP "\fB\-\-quiet\fR" 4
.IX Item "--quiet"
.PD
Turn off Wget's output.
.IP "\fB\-v\fR" 4
.IX Item "-v"
.PD 0
.IP "\fB\-\-verbose\fR" 4
.IX Item "--verbose"
.PD
Turn on verbose output, with all the available data.  The default output
is verbose.
.IP "\fB\-nv\fR" 4
.IX Item "-nv"
.PD 0
.IP "\fB\-\-no\-verbose\fR" 4
.IX Item "--no-verbose"
.PD
Turn off verbose without being completely quiet (use \fB\-q\fR for
that), which means that error messages and basic information still get
printed.
.IP "\fB\-\-report\-speed=\fR\fItype\fR" 4
.IX Item "--report-speed=type"
Output bandwidth as \fItype\fR.  The only accepted value is \fBbits\fR.
.IP "\fB\-i\fR \fIfile\fR" 4
.IX Item "-i file"
.PD 0
.IP "\fB\-\-input\-file=\fR\fIfile\fR" 4
.IX Item "--input-file=file"
.PD
Read URLs from a local or external \fIfile\fR.  If \fB\-\fR is
specified as \fIfile\fR, URLs are read from the standard input.  
(Use \fB./\-\fR to read from a file literally named \fB\-\fR.)
.Sp
If this function is used, no URLs need be present on the command
line.  If there are URLs both on the command line and in an input
file, those on the command lines will be the first ones to be
retrieved.  If \fB\-\-force\-html\fR is not specified, then \fIfile\fR
should consist of a series of URLs, one per line.
.Sp
However, if you specify \fB\-\-force\-html\fR, the document will be
regarded as \fBhtml\fR.  In that case you may have problems with
relative links, which you can solve either by adding \f(CW\*(C`<base
href="\f(CIurl\f(CW">\*(C'\fR to the documents or by specifying
\&\fB\-\-base=\fR\fIurl\fR on the command line.
.Sp
If the \fIfile\fR is an external one, the document will be automatically
treated as \fBhtml\fR if the Content-Type matches \fBtext/html\fR.
Furthermore, the \fIfile\fR's location will be implicitly used as base
href if none was specified.
.IP "\fB\-\-input\-metalink=\fR\fIfile\fR" 4
.IX Item "--input-metalink=file"
Downloads files covered in local Metalink \fIfile\fR. Metalink version 3
and 4 are supported.
.IP "\fB\-\-keep\-badhash\fR" 4
.IX Item "--keep-badhash"
Keeps downloaded Metalink's files with a bad hash. It appends .badhash
to the name of Metalink's files which have a checksum mismatch, except
without overwriting existing files.
.IP "\fB\-\-metalink\-over\-http\fR" 4
.IX Item "--metalink-over-http"
Issues \s-1HTTP HEAD\s0 request instead of \s-1GET\s0 and extracts Metalink metadata
from response headers. Then it switches to Metalink download.
If no valid Metalink metadata is found, it falls back to ordinary \s-1HTTP\s0 download.
Enables \fBContent-Type: application/metalink4+xml\fR files download/processing.
.IP "\fB\-\-metalink\-index=\fR\fInumber\fR" 4
.IX Item "--metalink-index=number"
Set the Metalink \fBapplication/metalink4+xml\fR metaurl ordinal
\&\s-1NUMBER.\s0 From 1 to the total number of \*(L"application/metalink4+xml\*(R"
available.  Specify 0 or \fBinf\fR to choose the first good one.
Metaurls, such as those from a \fB\-\-metalink\-over\-http\fR, may have
been sorted by priority key's value; keep this in mind to choose the
right \s-1NUMBER.\s0
.IP "\fB\-\-preferred\-location\fR" 4
.IX Item "--preferred-location"
Set preferred location for Metalink resources. This has effect if multiple
resources with same priority are available.
.IP "\fB\-F\fR" 4
.IX Item "-F"
.PD 0
.IP "\fB\-\-force\-html\fR" 4
.IX Item "--force-html"
.PD
When input is read from a file, force it to be treated as an \s-1HTML\s0
file.  This enables you to retrieve relative links from existing
\&\s-1HTML\s0 files on your local disk, by adding \f(CW\*(C`<base
href="\f(CIurl\f(CW">\*(C'\fR to \s-1HTML,\s0 or using the \fB\-\-base\fR command-line
option.
.IP "\fB\-B\fR \fI\s-1URL\s0\fR" 4
.IX Item "-B URL"
.PD 0
.IP "\fB\-\-base=\fR\fI\s-1URL\s0\fR" 4
.IX Item "--base=URL"
.PD
Resolves relative links using \fI\s-1URL\s0\fR as the point of reference,
when reading links from an \s-1HTML\s0 file specified via the
\&\fB\-i\fR/\fB\-\-input\-file\fR option (together with
\&\fB\-\-force\-html\fR, or when the input file was fetched remotely from
a server describing it as \s-1HTML\s0). This is equivalent to the
presence of a \f(CW\*(C`BASE\*(C'\fR tag in the \s-1HTML\s0 input file, with
\&\fI\s-1URL\s0\fR as the value for the \f(CW\*(C`href\*(C'\fR attribute.
.Sp
For instance, if you specify \fBhttp://foo/bar/a.html\fR for
\&\fI\s-1URL\s0\fR, and Wget reads \fB../baz/b.html\fR from the input file, it
would be resolved to \fBhttp://foo/baz/b.html\fR.
.IP "\fB\-\-config=\fR\fI\s-1FILE\s0\fR" 4
.IX Item "--config=FILE"
Specify the location of a startup file you wish to use.
.IP "\fB\-\-rejected\-log=\fR\fIlogfile\fR" 4
.IX Item "--rejected-log=logfile"
Logs all \s-1URL\s0 rejections to \fIlogfile\fR as comma separated values.  The values
include the reason of rejection, the \s-1URL\s0 and the parent \s-1URL\s0 it was found in.
.SS "Download Options"
.IX Subsection "Download Options"
.IP "\fB\-\-bind\-address=\fR\fI\s-1ADDRESS\s0\fR" 4
.IX Item "--bind-address=ADDRESS"
When making client \s-1TCP/IP\s0 connections, bind to \fI\s-1ADDRESS\s0\fR on
the local machine.  \fI\s-1ADDRESS\s0\fR may be specified as a hostname or \s-1IP\s0
address.  This option can be useful if your machine is bound to multiple
IPs.
.IP "\fB\-\-bind\-dns\-address=\fR\fI\s-1ADDRESS\s0\fR" 4
.IX Item "--bind-dns-address=ADDRESS"
[libcares only]
This address overrides the route for \s-1DNS\s0 requests. If you ever need to
circumvent the standard settings from /etc/resolv.conf, this option together
with \fB\-\-dns\-servers\fR is your friend.
\&\fI\s-1ADDRESS\s0\fR must be specified either as IPv4 or IPv6 address.
Wget needs to be built with libcares for this option to be available.
.IP "\fB\-\-dns\-servers=\fR\fI\s-1ADDRESSES\s0\fR" 4
.IX Item "--dns-servers=ADDRESSES"
[libcares only]
The given address(es) override the standard nameserver
addresses,  e.g. as configured in /etc/resolv.conf.
\&\fI\s-1ADDRESSES\s0\fR may be specified either as IPv4 or IPv6 addresses,
comma-separated.
Wget needs to be built with libcares for this option to be available.
.IP "\fB\-t\fR \fInumber\fR" 4
.IX Item "-t number"
.PD 0
.IP "\fB\-\-tries=\fR\fInumber\fR" 4
.IX Item "--tries=number"
.PD
Set number of tries to \fInumber\fR. Specify 0 or \fBinf\fR for
infinite retrying.  The default is to retry 20 times, with the exception
of fatal errors like \*(L"connection refused\*(R" or \*(L"not found\*(R" (404),
which are not retried.
.IP "\fB\-O\fR \fIfile\fR" 4
.IX Item "-O file"
.PD 0
.IP "\fB\-\-output\-document=\fR\fIfile\fR" 4
.IX Item "--output-document=file"
.PD
The documents will not be written to the appropriate files, but all
will be concatenated together and written to \fIfile\fR.  If \fB\-\fR
is used as \fIfile\fR, documents will be printed to standard output,
disabling link conversion.  (Use \fB./\-\fR to print to a file
literally named \fB\-\fR.)
.Sp
Use of \fB\-O\fR is \fInot\fR intended to mean simply "use the name
\&\fIfile\fR instead of the one in the \s-1URL\s0;" rather, it is
analogous to shell redirection:
\&\fBwget \-O file http://foo\fR is intended to work like
\&\fBwget \-O \- http://foo > file\fR; \fIfile\fR will be truncated
immediately, and \fIall\fR downloaded content will be written there.
.Sp
For this reason, \fB\-N\fR (for timestamp-checking) is not supported
in combination with \fB\-O\fR: since \fIfile\fR is always newly
created, it will always have a very new timestamp. A warning will be
issued if this combination is used.
.Sp
Similarly, using \fB\-r\fR or \fB\-p\fR with \fB\-O\fR may not work as
you expect: Wget won't just download the first file to \fIfile\fR and
then download the rest to their normal names: \fIall\fR downloaded
content will be placed in \fIfile\fR. This was disabled in version
1.11, but has been reinstated (with a warning) in 1.11.2, as there are
some cases where this behavior can actually have some use.
.Sp
A combination with \fB\-nc\fR is only accepted if the given output
file does not exist.
.Sp
Note that a combination with \fB\-k\fR is only permitted when
downloading a single document, as in that case it will just convert
all relative URIs to external ones; \fB\-k\fR makes no sense for
multiple URIs when they're all being downloaded to a single file;
\&\fB\-k\fR can be used only when the output is a regular file.
.IP "\fB\-nc\fR" 4
.IX Item "-nc"
.PD 0
.IP "\fB\-\-no\-clobber\fR" 4
.IX Item "--no-clobber"
.PD
If a file is downloaded more than once in the same directory, Wget's
behavior depends on a few options, including \fB\-nc\fR.  In certain
cases, the local file will be \fIclobbered\fR, or overwritten, upon
repeated download.  In other cases it will be preserved.
.Sp
When running Wget without \fB\-N\fR, \fB\-nc\fR, \fB\-r\fR, or
\&\fB\-p\fR, downloading the same file in the same directory will result
in the original copy of \fIfile\fR being preserved and the second copy
being named \fIfile\fR\fB.1\fR.  If that file is downloaded yet
again, the third copy will be named \fIfile\fR\fB.2\fR, and so on.
(This is also the behavior with \fB\-nd\fR, even if \fB\-r\fR or
\&\fB\-p\fR are in effect.)  When \fB\-nc\fR is specified, this behavior
is suppressed, and Wget will refuse to download newer copies of
\&\fIfile\fR.  Therefore, "\f(CW\*(C`no\-clobber\*(C'\fR" is actually a
misnomer in this mode\-\-\-it's not clobbering that's prevented (as the
numeric suffixes were already preventing clobbering), but rather the
multiple version saving that's prevented.
.Sp
When running Wget with \fB\-r\fR or \fB\-p\fR, but without \fB\-N\fR,
\&\fB\-nd\fR, or \fB\-nc\fR, re-downloading a file will result in the
new copy simply overwriting the old.  Adding \fB\-nc\fR will prevent
this behavior, instead causing the original version to be preserved
and any newer copies on the server to be ignored.
.Sp
When running Wget with \fB\-N\fR, with or without \fB\-r\fR or
\&\fB\-p\fR, the decision as to whether or not to download a newer copy
of a file depends on the local and remote timestamp and size of the
file.  \fB\-nc\fR may not be specified at the
same time as \fB\-N\fR.
.Sp
A combination with \fB\-O\fR/\fB\-\-output\-document\fR is only accepted
if the given output file does not exist.
.Sp
Note that when \fB\-nc\fR is specified, files with the suffixes
\&\fB.html\fR or \fB.htm\fR will be loaded from the local disk and
parsed as if they had been retrieved from the Web.
.IP "\fB\-\-backups=\fR\fIbackups\fR" 4
.IX Item "--backups=backups"
Before (over)writing a file, back up an existing file by adding a
\&\fB.1\fR suffix (\fB_1\fR on \s-1VMS\s0) to the file name.  Such backup
files are rotated to \fB.2\fR, \fB.3\fR, and so on, up to
\&\fIbackups\fR (and lost beyond that).
.IP "\fB\-c\fR" 4
.IX Item "-c"
.PD 0
.IP "\fB\-\-continue\fR" 4
.IX Item "--continue"
.PD
Continue getting a partially-downloaded file.  This is useful when you
want to finish up a download started by a previous instance of Wget, or
by another program.  For instance:
.Sp
.Vb 1
\&        wget \-c ftp://sunsite.doc.ic.ac.uk/ls\-lR.Z
.Ve
.Sp
If there is a file named \fIls\-lR.Z\fR in the current directory, Wget
will assume that it is the first portion of the remote file, and will
ask the server to continue the retrieval from an offset equal to the
length of the local file.
.Sp
Note that you don't need to specify this option if you just want the
current invocation of Wget to retry downloading a file should the
connection be lost midway through.  This is the default behavior.
\&\fB\-c\fR only affects resumption of downloads started \fIprior\fR to
this invocation of Wget, and whose local files are still sitting around.
.Sp
Without \fB\-c\fR, the previous example would just download the remote
file to \fIls\-lR.Z.1\fR, leaving the truncated \fIls\-lR.Z\fR file
alone.
.Sp
If you use \fB\-c\fR on a non-empty file, and the server does not support
continued downloading, Wget will restart the download from scratch and overwrite
the existing file entirely.
.Sp
Beginning with Wget 1.7, if you use \fB\-c\fR on a file which is of
equal size as the one on the server, Wget will refuse to download the
file and print an explanatory message.  The same happens when the file
is smaller on the server than locally (presumably because it was changed
on the server since your last download attempt)\-\-\-because \*(L"continuing\*(R"
is not meaningful, no download occurs.
.Sp
On the other side of the coin, while using \fB\-c\fR, any file that's
bigger on the server than locally will be considered an incomplete
download and only \f(CW\*(C`(length(remote) \- length(local))\*(C'\fR bytes will be
downloaded and tacked onto the end of the local file.  This behavior can
be desirable in certain cases\-\-\-for instance, you can use \fBwget \-c\fR
to download just the new portion that's been appended to a data
collection or log file.
.Sp
However, if the file is bigger on the server because it's been
\&\fIchanged\fR, as opposed to just \fIappended\fR to, you'll end up
with a garbled file.  Wget has no way of verifying that the local file
is really a valid prefix of the remote file.  You need to be especially
careful of this when using \fB\-c\fR in conjunction with \fB\-r\fR,
since every file will be considered as an \*(L"incomplete download\*(R" candidate.
.Sp
Another instance where you'll get a garbled file if you try to use
\&\fB\-c\fR is if you have a lame \s-1HTTP\s0 proxy that inserts a
\&\*(L"transfer interrupted\*(R" string into the local file.  In the future a
\&\*(L"rollback\*(R" option may be added to deal with this case.
.Sp
Note that \fB\-c\fR only works with \s-1FTP\s0 servers and with \s-1HTTP\s0
servers that support the \f(CW\*(C`Range\*(C'\fR header.
.IP "\fB\-\-start\-pos=\fR\fI\s-1OFFSET\s0\fR" 4
.IX Item "--start-pos=OFFSET"
Start downloading at zero-based position \fI\s-1OFFSET\s0\fR.  Offset may be expressed
in bytes, kilobytes with the `k' suffix, or megabytes with the `m' suffix, etc.
.Sp
\&\fB\-\-start\-pos\fR has higher precedence over \fB\-\-continue\fR.  When
\&\fB\-\-start\-pos\fR and \fB\-\-continue\fR are both specified, wget will emit a
warning then proceed as if \fB\-\-continue\fR was absent.
.Sp
Server support for continued download is required, otherwise \fB\-\-start\-pos\fR
cannot help.  See \fB\-c\fR for details.
.IP "\fB\-\-progress=\fR\fItype\fR" 4
.IX Item "--progress=type"
Select the type of the progress indicator you wish to use.  Legal
indicators are \*(L"dot\*(R" and \*(L"bar\*(R".
.Sp
The \*(L"bar\*(R" indicator is used by default.  It draws an \s-1ASCII\s0 progress
bar graphics (a.k.a \*(L"thermometer\*(R" display) indicating the status of
retrieval.  If the output is not a \s-1TTY,\s0 the \*(L"dot\*(R" bar will be used by
default.
.Sp
Use \fB\-\-progress=dot\fR to switch to the \*(L"dot\*(R" display.  It traces
the retrieval by printing dots on the screen, each dot representing a
fixed amount of downloaded data.
.Sp
The progress \fItype\fR can also take one or more parameters.  The parameters
vary based on the \fItype\fR selected.  Parameters to \fItype\fR are passed by
appending them to the type sperated by a colon (:) like this:
\&\fB\-\-progress=\fR\fItype\fR\fB:\fR\fIparameter1\fR\fB:\fR\fIparameter2\fR.
.Sp
When using the dotted retrieval, you may set the \fIstyle\fR by
specifying the type as \fBdot:\fR\fIstyle\fR.  Different styles assign
different meaning to one dot.  With the \f(CW\*(C`default\*(C'\fR style each dot
represents 1K, there are ten dots in a cluster and 50 dots in a line.
The \f(CW\*(C`binary\*(C'\fR style has a more \*(L"computer\*(R"\-like orientation\-\-\-8K
dots, 16\-dots clusters and 48 dots per line (which makes for 384K
lines).  The \f(CW\*(C`mega\*(C'\fR style is suitable for downloading large
files\-\-\-each dot represents 64K retrieved, there are eight dots in a
cluster, and 48 dots on each line (so each line contains 3M).
If \f(CW\*(C`mega\*(C'\fR is not enough then you can use the \f(CW\*(C`giga\*(C'\fR
style\-\-\-each dot represents 1M retrieved, there are eight dots in a
cluster, and 32 dots on each line (so each line contains 32M).
.Sp
With \fB\-\-progress=bar\fR, there are currently two possible parameters,
\&\fIforce\fR and \fInoscroll\fR.
.Sp
When the output is not a \s-1TTY,\s0 the progress bar always falls back to \*(L"dot\*(R",
even if \fB\-\-progress=bar\fR was passed to Wget during invocation. This
behaviour can be overridden and the \*(L"bar\*(R" output forced by using the \*(L"force\*(R"
parameter as \fB\-\-progress=bar:force\fR.
.Sp
By default, the \fBbar\fR style progress bar scroll the name of the file from
left to right for the file being downloaded if the filename exceeds the maximum
length allotted for its display.  In certain cases, such as with
\&\fB\-\-progress=bar:force\fR, one may not want the scrolling filename in the
progress bar.  By passing the \*(L"noscroll\*(R" parameter, Wget can be forced to
display as much of the filename as possible without scrolling through it.
.Sp
Note that you can set the default style using the \f(CW\*(C`progress\*(C'\fR
command in \fI.wgetrc\fR.  That setting may be overridden from the
command line.  For example, to force the bar output without scrolling,
use \fB\-\-progress=bar:force:noscroll\fR.
.IP "\fB\-\-show\-progress\fR" 4
.IX Item "--show-progress"
Force wget to display the progress bar in any verbosity.
.Sp
By default, wget only displays the progress bar in verbose mode.  One may
however, want wget to display the progress bar on screen in conjunction with
any other verbosity modes like \fB\-\-no\-verbose\fR or \fB\-\-quiet\fR.  This
is often a desired a property when invoking wget to download several small/large
files.  In such a case, wget could simply be invoked with this parameter to get
a much cleaner output on the screen.
.Sp
This option will also force the progress bar to be printed to \fIstderr\fR when
used alongside the \fB\-\-logfile\fR option.
.IP "\fB\-N\fR" 4
.IX Item "-N"
.PD 0
.IP "\fB\-\-timestamping\fR" 4
.IX Item "--timestamping"
.PD
Turn on time-stamping.
.IP "\fB\-\-no\-if\-modified\-since\fR" 4
.IX Item "--no-if-modified-since"
Do not send If-Modified-Since header in \fB\-N\fR mode. Send preliminary \s-1HEAD\s0
request instead. This has only effect in \fB\-N\fR mode.
.IP "\fB\-\-no\-use\-server\-timestamps\fR" 4
.IX Item "--no-use-server-timestamps"
Don't set the local file's timestamp by the one on the server.
.Sp
By default, when a file is downloaded, its timestamps are set to
match those from the remote file. This allows the use of
\&\fB\-\-timestamping\fR on subsequent invocations of wget. However, it
is sometimes useful to base the local file's timestamp on when it was
actually downloaded; for that purpose, the
\&\fB\-\-no\-use\-server\-timestamps\fR option has been provided.
.IP "\fB\-S\fR" 4
.IX Item "-S"
.PD 0
.IP "\fB\-\-server\-response\fR" 4
.IX Item "--server-response"
.PD
Print the headers sent by \s-1HTTP\s0 servers and responses sent by
\&\s-1FTP\s0 servers.
.IP "\fB\-\-spider\fR" 4
.IX Item "--spider"
When invoked with this option, Wget will behave as a Web \fIspider\fR,
which means that it will not download the pages, just check that they
are there.  For example, you can use Wget to check your bookmarks:
.Sp
.Vb 1
\&        wget \-\-spider \-\-force\-html \-i bookmarks.html
.Ve
.Sp
This feature needs much more work for Wget to get close to the
functionality of real web spiders.
.IP "\fB\-T seconds\fR" 4
.IX Item "-T seconds"
.PD 0
.IP "\fB\-\-timeout=\fR\fIseconds\fR" 4
.IX Item "--timeout=seconds"
.PD
Set the network timeout to \fIseconds\fR seconds.  This is equivalent
to specifying \fB\-\-dns\-timeout\fR, \fB\-\-connect\-timeout\fR, and
\&\fB\-\-read\-timeout\fR, all at the same time.
.Sp
When interacting with the network, Wget can check for timeout and
abort the operation if it takes too long.  This prevents anomalies
like hanging reads and infinite connects.  The only timeout enabled by
default is a 900\-second read timeout.  Setting a timeout to 0 disables
it altogether.  Unless you know what you are doing, it is best not to
change the default timeout settings.
.Sp
All timeout-related options accept decimal values, as well as
subsecond values.  For example, \fB0.1\fR seconds is a legal (though
unwise) choice of timeout.  Subsecond timeouts are useful for checking
server response times or for testing network latency.
.IP "\fB\-\-dns\-timeout=\fR\fIseconds\fR" 4
.IX Item "--dns-timeout=seconds"
Set the \s-1DNS\s0 lookup timeout to \fIseconds\fR seconds.  \s-1DNS\s0 lookups that
don't complete within the specified time will fail.  By default, there
is no timeout on \s-1DNS\s0 lookups, other than that implemented by system
libraries.
.IP "\fB\-\-connect\-timeout=\fR\fIseconds\fR" 4
.IX Item "--connect-timeout=seconds"
Set the connect timeout to \fIseconds\fR seconds.  \s-1TCP\s0 connections that
take longer to establish will be aborted.  By default, there is no
connect timeout, other than that implemented by system libraries.
.IP "\fB\-\-read\-timeout=\fR\fIseconds\fR" 4
.IX Item "--read-timeout=seconds"
Set the read (and write) timeout to \fIseconds\fR seconds.  The
\&\*(L"time\*(R" of this timeout refers to \fIidle time\fR: if, at any point in
the download, no data is received for more than the specified number
of seconds, reading fails and the download is restarted.  This option
does not directly affect the duration of the entire download.
.Sp

wget/1.19.1_1/.brew/wget.rb
# NOTE: Configure will fail if using awk 20110810 from dupes.
# Upstream issue: https://savannah.gnu.org/bugs/index.php?37063
class Wget < Formula
  desc "Internet file retriever"
  homepage "https://www.gnu.org/software/wget/"
  url "https://ftp.gnu.org/gnu/wget/wget-1.19.1.tar.gz"
  mirror "https://ftpmirror.gnu.org/wget/wget-1.19.1.tar.gz"
  sha256 "9e4f12da38cc6167d0752d934abe27c7b1599a9af294e73829be7ac7b5b4da40"
  revision 1

  head do
    url "https://git.savannah.gnu.org/git/wget.git"

    depends_on "autoconf" => :build
    depends_on "automake" => :build
    depends_on "xz" => :build
    depends_on "gettext"
  end

  deprecated_option "enable-debug" => "with-debug"

  option "with-debug", "Build with debug support"

  depends_on "pkg-config" => :build
  depends_on "pod2man" => :build if MacOS.version <= :snow_leopard
  depends_on "openssl@1.1"
  depends_on "pcre" => :optional
  depends_on "libmetalink" => :optional
  depends_on "gpgme" => :optional

  def install
    # Fixes undefined symbols _iconv, _iconv_close, _iconv_open
    # Reported 10 Jun 2016: https://savannah.gnu.org/bugs/index.php?48193
    ENV.append "LDFLAGS", "-liconv"

    args = %W[
      --prefix=#{prefix}
      --sysconfdir=#{etc}
      --with-ssl=openssl
      --with-libssl-prefix=#{Formula["openssl@1.1"].opt_prefix}
    ]

    args << "--disable-debug" if build.without? "debug"
    args << "--disable-pcre" if build.without? "pcre"
    args << "--with-metalink" if build.with? "libmetalink"
    args << "--with-gpgme-prefix=#{Formula["gpgme"].opt_prefix}" if build.with? "gpgme"

    system "./bootstrap" if build.head?
    system "./configure", *args
    system "make", "install"
  end

  test do
    system bin/"wget", "-O", "/dev/null", "https://google.com"
  end
end
